src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,88 @@
+package lotto.controller;
+
+import java.util.List;
+import lotto.domain.amount.PurchaseAmount;
+import lotto.domain.lotto.BonusNumber;
+import lotto.domain.lotto.LottoMachine;
+import lotto.domain.lotto.LottoStatistic;
+import lotto.domain.lotto.Lottos;
+import lotto.domain.lotto.WinningLotto;
+import lotto.dto.LottosResponse;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class LottoController {
+
+    private final LottoMachine lottoMachine;
+    private final LottoStatistic lottoStatistic;
+
+    public LottoController() {
+        this.lottoMachine = LottoMachine.initial();
+        this.lottoStatistic = LottoStatistic.initial();
+    }
+
+    public void run() {
+        PurchaseAmount purchaseAmount = receivePurchaseAmount();
+        Lottos lottos = purchaseLottos(purchaseAmount);
+        WinningLotto winningLotto = receiveWinningLotto();
+        computeStatistics(lottos, winningLotto, purchaseAmount);
+    }
+
+    private PurchaseAmount receivePurchaseAmount() {
+        while (true) {
+            try {
+                int amount = InputView.readPurchaseAmount();
+                return PurchaseAmount.from(amount);
+            } catch (IllegalArgumentException e) {
+                OutputView.printError(e.getMessage());
+            }
+        }
+    }
```

Comment:
```suggestion
    private PurchaseAmount receivePurchaseAmount() {
            try {
                int amount = InputView.readPurchaseAmount();
                return PurchaseAmount.from(amount);
            } catch (IllegalArgumentException e) {
                OutputView.printError(e.getMessage());
                receivePurchaseAmount();
            }
        }
    }
```
재귀를 사용한다면 반복문을 제거해줄 수 있어서 가독성에 더 좋을 것 같아요!
물론 성능상은 조금 안좋지만 가독성 측면에서 이득을 취할 수 있습니다! 😁

Reply:
> 재귀를 사용한다면 반복문을 제거해줄 수 있어서 가독성에 더 좋을 것 같아요! 물론 성능상은 조금 안좋지만 가독성 측면에서 이득을 취할 수 있습니다! 😁

좋은 말씀 감사합니다 : ) 사실 저도 처음에는 재귀로 구현했는데요, 자바 특성 상 1000번이 넘어가면 StackOverFlow가 발생하더라구요! 여러 예외 상황에 대해서 깊게 생각하다 보니 자연스럽게 while로 구현하게 되었습니다! 

좋은 의견 감사합니다~!

Reply:
저도 재귀로 구현하였는데 StackOverFlow 까지는 생각을 못하고 그냥 구현을 했네요..!
이번 기회에 공부해봐야겠습니다

Reply:
저도 StackOverFlow는 생각을 못했습니다 😮 구현한 내용에 어떤 예외사항이 생길지 깊이 고민하고 작성해야겠네요

Reply:
저도 StackOverFlow는 생각하지 못했네요. 재귀를 사용할 때 이러한 부분도 간과해서는 안될 것 같네요

Reply:
저도 궁금했던 부분인데 하나 배워가네요..! 
좋은 정보 감사합니다ㅎ